packages/cli/generators/discover/index.js

```diff
@@ -321,6 +328,32 @@ module.exports = class DiscoveryGenerator extends ArtifactGenerator {
       );
       debug(`Writing: ${fullPath}`);
 
+      if (this.options.relations) {
+        const relationImports = [];
+        const relationDestinationImports = [];
+        for (const relationName in templateData.settings.relations) {
+          const relation = templateData.settings.relations[relationName];
+          const targetModel = this.artifactInfo.modelDefinitions.find(
+            model => model.name === relation.model,
+          );
+          // If targetModel is not in discovered models, skip creating relation
+          if (targetModel) {
+            Object.assign(templateData.properties[relation.foreignKey], {
+              relation,
+            });
+            relationImports.push(relation.type);
+            relationDestinationImports.push(relation.model);
+          }
+        }
+        templateData.relationImports = relationImports;
+        templateData.relationDestinationImports = relationDestinationImports;
+        // Delete relation from modelSettings
+        delete templateData.settings.relations;
```

Comment:
Some connectors support "strong relations" (i.e. Constraints enforced by the database engine) through `@model({settings: {foreignKeys: {/* ... */}})`. Can we populate that field as well?

See https://github.com/loopbackio/loopback-connector-mysql/tree/f8f40a1199b6ed63ba01d6d173a9314004c08c93#auto-migrateauto-update-models-with-foreign-keys for an example of the syntax.

Reply:
More info can be found here: https://github.com/loopbackio/loopback-datasource-juggler/blob/055efd3e89e2326bc81e348cbb1c6ef973ab6428/types/model.d.ts#L160-L169

...which is part of this PR: https://github.com/loopbackio/loopback-datasource-juggler/pull/1904

Reply:
@achrinza  Thank you for the review. and sure we can do that.

Reply:
@achrinza, we have updated the code to populate `foreignKeys` in model settings.

Reply:
Hi @achrinza, 
The Soap-calculator example uses a soap service link `https://calculator-webservice.mybluemix.net/` for sending HTTP requests. This link is now returning 404 on all requests and because of that related tests are failing. We are unable to get our PRs merged because of these failed tests.
This is the link for datasource of soap-service.
https://github.com/loopbackio/loopback-next/blob/master/examples/soap-calculator/src/datasources/calculator.datasource.ts

Reply:
Hi @sohaibahsan007, we're aware of the issue and looking into it.

Reply:
Thank you.